### ISSUE TITLE
huber loss, ema and flow shift

### ISSUE THREAD

**Issue by veroveroxie** [NVlabs/Sana#191]
Hi, 

Thanks for releasing this wonderful source code.  It is really impressive. I tried to follow your code. But I may some questions related to your config:

1. Huber loss, in your TrainingConfig, you defined the loss to be huber loss, but I cannot find where it is being used.  The loss types used seem to be MSE? `if use_kl:
        loss_type = gd.LossType.RESCALED_KL
    elif rescale_learned_sigmas:
        loss_type = gd.LossType.RESCALED_MSE
    else:
        loss_type = gd.LossType.MSE`

2. EMA. I noticed that you used EMA with ema_rate 0.9999 and ema_decay 0.95 and an update_ema function. However, it seems not being used in train.py, could you please tell me is EMA used during training or not? 

3.  I am particularly interested in training 512px model due to computing resources. I found that the config uses flow shift as 1.0 while other models use flow shift as 3.0. May I know that is it a typo or it is carefully selected for 512px model?

Thank you so much!

**Comment by lawrence-cj** [comment]
> Huber loss, in your TrainingConfig, you defined the loss to be huber loss, but I cannot find where it is being used. The loss types used seem to be MSE? if use_kl: loss_type = gd.LossType.RESCALED_KL elif rescale_learned_sigmas: loss_type = gd.LossType.RESCALED_MSE else: loss_type = gd.LossType.MSE

You are correct. There is no huber loss in our diffusion training. 

> EMA. I noticed that you used EMA with ema_rate 0.9999 and ema_decay 0.95 and an update_ema function. However, it seems not being used in train.py, could you please tell me is EMA used during training or not?

We just released the EMA related code, but we don't use it during our pre-training.


> I am particularly interested in training 512px model due to computing resources. I found that the config uses flow shift as 1.0 while other models use flow shift as 3.0. May I know that is it a typo or it is carefully selected for 512px model?

I'm afraid this one is a typo. Thank you for noticing, I will fix it in our next update.
